src/Range.tsx

```diff
@@ -16,7 +16,13 @@ import {
 } from './utils';
 import { IProps, Direction } from './types';
 
-class Range extends React.Component<IProps> {
+interface IState {
+  draggedThumbIndex: number;
+  thumbZIndexes: number[];
+  isChanged?: boolean;
```

Comment:
Why is there a need to detect this? I would expect `onDragEnd` to always fire.

If really needed, `isChanged: false` can be used as the initial state and no need to define `IState` interface.

Reply:
The point is that not every keyup event should fire `onFinalUpdate` event e.g when we pressed the `Tab` button.
But you are right that it could be in an initial state without defining an interface

Reply:
Oh, I get it now. Well, it would be better to just filter key events the same way as it is done in `onKeyDown` (ideally, extract those filters/conditions into so it's in a single place). No need to introduce an another state, right?

Reply:
I've found some issue with this case and offer to switch back to the state:
When you press some arrow key, thumb moved, value changed. But if you press another key, e.g. `Tab` without raising arrow key, we lose a focus and `onFinalChange` won't be fired.

Reply:
Fixed this and added test.